src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,137 @@
+     */
+    public static Command parse(String input) throws DukeException, IllegalArgumentException,
+            DateTimeParseException, NumberFormatException {
+
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        }
+
+        // Split strings into 2, first part is the operation, 2nd part is the description
+        String[] command = input.split(" ", 2);
+        Operation op = Operation.valueOf(command[0].toUpperCase());
+
+        if (!op.equals(Operation.LIST) && command.length < 2) {
+            throw new DukeException("No command description given.");
+        }
+
+        switch (op) {
+        case MARK:
+            return markTaskParser(input);
+        case UNMARK:
+            return unmarkTaskParser(input);
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            return addTaskParser(input);
+        case LIST:
+            return new ListTasksCommand();
+        case DELETE:
+            return deleteTaskParser(input);
+        case FIND:
+            return findTaskParser(input);
+        }
+
+        return null; // cannot reach here, as duke.command.Operation.valueOf throws IllegalArgumentException
+    }
+
+    /**
+     * Parses the mark command input.
+     *
+     * @param input User input.
+     * @return MarkTaskCommand instance.
+     * @throws NumberFormatException If format of task index is invalid.
+     */
+    public static Command markTaskParser(String input) throws NumberFormatException {
+        String[] command = input.split(" ", 2);
+        int taskIndex = Integer.parseInt(command[1]); // Throws exception if not valid string as id
+        return new MarkTaskCommand(taskIndex);
+    }
+
+    /**
+     * Parses the unmark command input.
+     *
+     * @param input User input.
+     * @return UnmarkTaskCommand instance.
+     * @throws NumberFormatException If format of task index is invalid.
+     */
+    public static Command unmarkTaskParser(String input) throws NumberFormatException {
+        String[] command = input.split(" ", 2);
+        int taskIndex = Integer.parseInt(command[1]); // Throws exception if not valid string as id
+        return new UnmarkTaskCommand(taskIndex);
+    }
+
+    /**
+     * Parses the command to add a task.
+     * @param input User input.
+     * @return A Command instance representing the type of task to add.
+     * @throws DateTimeParseException If the date format is invalid.
+     */
+    public static Command addTaskParser(String input) throws DateTimeParseException {
+        String[] command = input.split(" ", 2);
+        Operation op = Operation.valueOf(command[0].toUpperCase());
+        String description = command[1];
+
+        switch (op) {
+        case TODO:
+            return new AddTodoCommand(description);
+        case DEADLINE:
+            String[] deadlineString = description.split("/by", 2);
+            String deadlineDescription = deadlineString[0];
```

Comment:
Perhaps `trim()` the description since there might be a `' '` before the `/by`?

Reply:
This makes sense, thank you!